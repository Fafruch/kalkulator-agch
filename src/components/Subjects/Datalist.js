import React from 'react'
import PropTypes from 'prop-types'

import { PRIMARY_SUBJECTS } from '../../constants/SubjectTypes'

const Datalist = ({ subjectsType }) => {
  if (subjectsType === PRIMARY_SUBJECTS) {
    return (
      <datalist id={subjectsType}>
        <option value='Biologia' />
        <option value='Chemia' />
        <option value='Fizyka' />
        <option value='Geografia' />
        <option value='Informatyka' />
        <option value='Matematyka' />

        <option value='Filozofia' />
        <option value='Historia' />
        <option value='Historia sztuki' />
        <option value='Wiedza o społeczeństwie' />
        <option value='Język polski' />
      </datalist>
    )
  }
  return (
    <datalist id={subjectsType}>
      <option value='Język angielski' />
      <option value='Język francuski' />
      <option value='Język hiszpański' />
      <option value='Język niemiecki' />
      <option value='Język rosyjski' />
      <option value='Język włoski' />
    </datalist>
  )
}
Datalist.propTypes = {
  subjectsType: PropTypes.string.isRequired
}

export default Datalist
