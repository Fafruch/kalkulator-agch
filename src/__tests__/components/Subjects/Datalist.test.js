import React from 'react'
import { shallow } from 'enzyme'

import Datalist from '../../../components/Subjects/Datalist'

const setup = (setupProps = {}) => {
  const defaultProps = {
    subjectsType: 'primary'
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<Datalist {...props} />)

  return {
    props,
    wrapper
  }
}

describe('Datalist component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })

  test('should render datalist with id attribute of the subjectsType', () => {
    let { wrapper } = setup()

    expect(wrapper.prop('id')).toEqual('primary')

    ;({ wrapper } = setup({ subjectsType: 'lingual' }))

    expect(wrapper.prop('id')).toEqual('lingual')
  })
})
