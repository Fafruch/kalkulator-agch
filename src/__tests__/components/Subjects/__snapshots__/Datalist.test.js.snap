// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Datalist component should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <datalist
    id="primary"
>
    <option
        value="Biologia"
    />
    <option
        value="Chemia"
    />
    <option
        value="Fizyka"
    />
    <option
        value="Geografia"
    />
    <option
        value="Informatyka"
    />
    <option
        value="Matematyka"
    />
    <option
        value="Filozofia"
    />
    <option
        value="Historia"
    />
    <option
        value="Historia sztuki"
    />
    <option
        value="Wiedza o społeczeństwie"
    />
    <option
        value="Język polski"
    />
</datalist>,
  "nodes": Array [
    <datalist
      id="primary"
>
      <option
            value="Biologia"
      />
      <option
            value="Chemia"
      />
      <option
            value="Fizyka"
      />
      <option
            value="Geografia"
      />
      <option
            value="Informatyka"
      />
      <option
            value="Matematyka"
      />
      <option
            value="Filozofia"
      />
      <option
            value="Historia"
      />
      <option
            value="Historia sztuki"
      />
      <option
            value="Wiedza o społeczeństwie"
      />
      <option
            value="Język polski"
      />
</datalist>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Datalist
        subjectsType="primary"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "subjectsType": "primary",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <datalist
          id="primary"
>
          <option
                    value="Biologia"
          />
          <option
                    value="Chemia"
          />
          <option
                    value="Fizyka"
          />
          <option
                    value="Geografia"
          />
          <option
                    value="Informatyka"
          />
          <option
                    value="Matematyka"
          />
          <option
                    value="Filozofia"
          />
          <option
                    value="Historia"
          />
          <option
                    value="Historia sztuki"
          />
          <option
                    value="Wiedza o społeczeństwie"
          />
          <option
                    value="Język polski"
          />
</datalist>,
        "_debugID": 2,
        "_renderedOutput": <datalist
          id="primary"
>
          <option
                    value="Biologia"
          />
          <option
                    value="Chemia"
          />
          <option
                    value="Fizyka"
          />
          <option
                    value="Geografia"
          />
          <option
                    value="Informatyka"
          />
          <option
                    value="Matematyka"
          />
          <option
                    value="Filozofia"
          />
          <option
                    value="Historia"
          />
          <option
                    value="Historia sztuki"
          />
          <option
                    value="Wiedza o społeczeństwie"
          />
          <option
                    value="Język polski"
          />
</datalist>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Datalist
    subjectsType="primary"
/>,
}
`;
